Halachic Times/app/src/main/java/com/github/times/ZmanimDetailsPopulater.java

```diff
@@ -961,18 +1011,208 @@ private void populateLatestKiddushLevana(A adapter, ComplexZmanimCalendar cal, Z
     }
 
     private void populateShabbathEnds(A adapter, ComplexZmanimCalendar cal, ZmanimPreferences settings) {
-        int endsAfter = settings.getShabbathEndsAfter();
         long offset = settings.getShabbathEnds() * MINUTE_IN_MILLIS;
-        switch (endsAfter) {
-            case R.string.sunset:
-                populateSunset(adapter, cal, settings, offset);
-                break;
-            case R.string.twilight:
-                populateTwilight(adapter, cal, settings, offset);
-                break;
-            case R.string.nightfall:
-                populateNightfall(adapter, cal, settings, offset);
-                break;
+        Long date;
```

Comment:
please move all the "shabbat ends" to another pull request separate from the main Baal haTanya changes.